src/Type/CollectionType.php

```diff
@@ -29,7 +29,7 @@ public function buildForm(FormBuilderInterface $builder, array $options): void
             $options['type'],
             $options['type_options'],
             $options['modifiable'],
-            $options['pre_bind_data_callback']
+            $options['modifiable']
```

Comment:
How does this relate to this PR?

Reply:
Symfony splits the `modifiable` options into `allow_add` and `allow_delete`.
And they doesn't support the `pre_bind_data_callback` option.

I still need some work on this PR, I'm not sure I'll be able to do this on 1.x since dropping the `pre_bind_data_callback` will be a BC-break.